content/kb/troubleshooting-orphan-instance.md

```diff
@@ -0,0 +1,108 @@
+---
+title: "Troubleshooting: Orphan Engine Or Replica Instance"
+authors:
+- "Raphanus Lo"
+draft: false
+date: 2025-05-19
+versions:
+- "All versions"
+categories:
+- "orphan"
+---
+
+## Applicable versions
+
+* All longhorn versions
+
+## Symptoms
+
+The `instancemanager` custom resource lists instances (including engine and replica instances), but the corresponding engine or replica objects no longer exist in the cluster.
+
+## Details
+
+During unexpected node disconnections or interruptions (such as during a Longhorn upgrade), engine or replica runtime instances can become detached from their corresponding custom resources. These are known as **orphaned instances**.
+
+The Instance Manager can detect orphaned instances and will list them in its status. However, it cannot remove them through standard cleanup procedures. This can block normal operations like system upgrades or node replacements, since the Instance Manager cannot be shut down, restarted, or upgraded until all managed instances are safely cleaned up.
+
+Starting with Longhorn v1.9.0, an orphaned instance tracking mechanism is introduced to help manage these resources. However, if the Instance Manager itself is too outdated, it may not be able to track or remove orphaned instances. This limitation can further complicate maintenance and upgrade processes.
```

Comment:
Not really sure what "This limitation can further complicate maintenance and upgrade processes."  means here.

```suggestion
Starting with Longhorn v1.9.0, an orphaned instance tracking mechanism is introduced to help manage these orphans. However, if the Instance Manager is created by Longhorn before v1.9.0, it is not able to track or remove orphaned instances. 
```

Reply:
Need to mention what the purpose of the following workaround is.